### DISCUSSION TITLE
Best Practices for parsing json items to dto

### DISCUSSION THREAD

**Discussion by thedangler** [saloonphp/saloon#378] [Q&A] [answered]
Hello,
I'm using saloon to get orders from an API and it returns a list of orders I'm wondering where I should parse the json?
Currently I'm using createDtoFromResponse for my parsing logic.

I'm looking for a better place to place the logic and the json response is also coming from a webhook in the same structure.

I haven't seen an example on how to turn large json response to dto.

Should I have a DTO for single Order and a DTO for Orders?

Regards

**Comment by Sammyjo20** [comment, marked as answer]
Hey @thedangler! 

When it comes to parsing lots of items, I tend to return an array in the `createDtoFromResponse` method and use `array_map` to convert each item into an individual DTO. This way there's a small amount of code and it becomes an array of DTOs.

```php
public function createDtoFromResponse(Response $response): array
{
    $items = $response->json(); // Get the array of items

    return array_map(function (array $item) { 
        return Order::fromArray($item);
    }, $items);
}
```

This would be how I would write the `Order` DTO.

```php
class Order
{
    public function __construct(
        public string $property, // properties...
    )

    public static function fromArray(array $data): self
    {
      return new self(
          property: $data['property']
      );
    }
}
```

Then when I call `$response->dto()` I would get an array of `Order` DTOs.

Does that help?

**Reply by thedangler** [reply]
Yes that helps. There is other logic I need to take care of too I was hoping to place somewhere else such that I don't have the same code in multiple places but it might not be avoidable. 

I have a goofy idea to make an internal call as a connector that gets the webhook data from my DB.

Thank you..
